src/dotdirdeps.cpp

```diff
@@ -20,6 +20,36 @@
 #include "doxygen.h"
 #include "config.h"
 
+
+
+static std::size_t getMaxDirectoryDepth()
+{
+  return 3;  //! @todo use a parameter for the max depth
+}
```

Comment:
Question / thoughts will this be used for:
- the parent direction
- the children direction
- half for parent, half for children and 1 for current (so must be odd)

Maybe there should be the possibility in the configuration file to specify
- a configuration variable for parent and one for children
- one configuration variable  according to "half" option as mentioned above
- a configurtion variable specfying both like `VAR=(parnet, children)`
- ...

Reply:
Good question. I was thinking of the following levels: 1 parent + 1 current + (x>0) configurable sub levels. Configuration would be probably done with `Config_getInt`? My intention was not to bring to many new parameters to the configuration. Thus imitating the current behaviour and only adding additional levels to the depth (sub-directories). I am grateful for feedback, thank you!

Reply:
The number of configuration parameters should indeed be limited (there are already quite a few), but I think that one fo parent and one for children would be best. In the code best is probably to have separate getMaxDirectoryDepth for parent and children so the flexibility is at least present in the code.

Reply:
I have defined only one configuration variable `MAX_DOT_GRAPH_SUCCESSOR` at the moment. IMHO the average use case is to inspect the directory structure and the dependencies for one specific directory. In case the view from a sup-directory / ancestor director is of interest, the user may select that directory instead.

I would like to leave it to future work to add a configuration variable for the number of ancestor levels - if it is of interest.